src/transformers/modeling_prophetnet.py

```diff
@@ -986,6 +723,50 @@ def ngram_attention_bias(length, num_skip):
             bias_result.append(bias_n_skip)
         return torch.from_numpy(np.array(bias_result, dtype=np.float32))
 
+def _relative_positions_bucket(num_buckets, max_distance, relative_positions, is_bidirectional=False):
+    n = -relative_positions
+    result = 0
+    if is_bidirectional:
+        num_buckets = num_buckets // 2
+        result = result + torch.lt(n, torch.zeros_like(n)).int() * num_buckets
+        n = torch.abs(n)
+    else:
+        n = torch.max(n, torch.zeros_like(n))
+    max_exact = num_buckets // 2
+    is_small = torch.lt(n, max_exact)
+    val_if_large = max_exact + torch.log(n.float() / max_exact) / math.log(max_distance / max_exact) * (
+            num_buckets - max_exact)
+    val_if_large = torch.min(val_if_large, torch.ones_like(val_if_large) * (num_buckets - 1))
+    val_if_large = val_if_large.int()
+    result = result + torch.where(is_small, n.int(), val_if_large)
+    return result
+
+
+def cal_relative_positions_buckets(num_buckets, max_distance, real_positions):
```

Comment:
I'm a little confused with the names of `_relative_positions_bucket` and `cal_relative_positions_buckets`.